api/agent/data_access.go

```diff
@@ -62,16 +62,16 @@ func appCacheKey(appname string) string {
 	return "a:" + appname
 }
 
-func (da *CachedDataAccess) GetApp(ctx context.Context, appName string) (*models.App, error) {
-	key := appCacheKey(appName)
+func (da *CachedDataAccess) GetApp(ctx context.Context, a *models.App) (*models.App, error) {
+	key := appCacheKey(a.Name)
```

Comment:
this should be by id. if a user renames the app and another user takes the same name, they'll get the first user's config out of the cache. this would be pretty bad